aiplatform/src/main/java/aiplatform/imagen/EditImageMaskFreeSample.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aiplatform.imagen;
+
+// [START generativeaionvertexai_imagen_edit_image_mask_free]
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.aiplatform.v1.EndpointName;
+import com.google.cloud.aiplatform.v1.PredictResponse;
+import com.google.cloud.aiplatform.v1.PredictionServiceClient;
+import com.google.cloud.aiplatform.v1.PredictionServiceSettings;
+import com.google.gson.Gson;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EditImageMaskFreeSample {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project-id";
+    String location = "us-central1";
+    String inputPath = "/path/to/my-input.png";
+    String prompt = ""; // The text prompt describing what you want to see.
+
+    editImageMaskFree(projectId, location, inputPath, prompt);
+  }
+
+  // Edit an image without using a mask. The edit is applied to the entire image and is saved to a
+  // new file.
+  public static PredictResponse editImageMaskFree(
+      String projectId, String location, String inputPath, String prompt)
+      throws ApiException, IOException {
+    final String endpoint = String.format("%s-aiplatform.googleapis.com:443", location);
+    PredictionServiceSettings predictionServiceSettings =
+        PredictionServiceSettings.newBuilder().setEndpoint(endpoint).build();
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (PredictionServiceClient predictionServiceClient =
+        PredictionServiceClient.create(predictionServiceSettings)) {
+
+      final EndpointName endpointName =
+          EndpointName.ofProjectLocationPublisherModelName(
+              projectId, location, "google", "imagegeneration@002");
```

Comment:
We're not using Imagen v3 here, right?
I guess it's because image editing is not yet available on our newer model?

Reply:
Correct, mask-free editing is v002 only: https://cloud.google.com/vertex-ai/generative-ai/docs/image/edit-images#model-versions-mask-free